tests/pytests/test_ddx.py

```diff
@@ -0,0 +1,250 @@
+       "geom": __geoms["h"],
+       "dm": core.Matrix.from_array(np.array([[0.]])),
+       "radii": [1.0],  # Angstrom
+       "ref": -0.2645886054599999,  # from Gaussian
+   }, id='h'),
+   pytest.param({
+       "geom": __geoms["h2"],
+       "dm": core.Matrix.from_array(0.6682326961201372 * np.ones((2, 2))),
+       "radii": [1.5873, 1.5873],  # Angstrom
+       "ref": -0.0002016948,  # from Gaussian
+   }, id='h2'),
+])
+def test_ddx_fock_build(inp):
+    """
+    Tests COSMO energy against reference from Gaussian
+    and internal consistency of Fock matrix versus the energy.
+    """
+    from psi4.driver.procrouting.solvent import ddx
+
+    psi4.set_options({
+        "df_scf_guess": False,
+        #
+        "ddx__model": "cosmo",
+        "ddx__solvent_epsilon": 1e8,
+        "ddx__eta": 0,
+        "ddx__lmax": 3,
+        "ddx__n_lebedev": 302,
+        "ddx__radii": inp["radii"],
+        "ddx__dft_spherical_points": 350,
+        "ddx__dft_radial_points": 99,
+    })
+
+    # build the DDX object to test
+    mol = psi4.geometry(inp["geom"])
+    basis = psi4.core.BasisSet.build(mol, "BASIS", "sto-3g")
+    ddx_options = ddx.get_ddx_options(mol)
+    ddx_state = ddx.DdxInterface(mol, ddx_options, basis)
+
+    def get_EV(density_matrix):
+        return ddx_state.get_solvation_contributions(density_matrix)
+
+    _base_test_fock(get_EV, inp["dm"])
+
+    E, _ = get_EV(inp["dm"])
+    assert compare_values(inp["ref"], E, atol=1e-16, rtol=1e-2)
+
+@pytest.mark.quick
+@uusing("ddx")
+@pytest.mark.parametrize("inp", [
+    pytest.param({
+        "geom": __geoms["h2o"],
+        "ddx": {"solvent_epsilon": 1e8, "eta": 0, },
+        "ref": -75.5946789010,  # from Gaussian
+        "solvation": -0.009402,
+    }, id='h2o'),
+    #
+    pytest.param({
+        "geom": __geoms["fcm"],
+        "ddx": {"solvent_epsilon": 1e8, "eta": 0, },
+        "ref": -594.993575419,  # from Gaussian
+        "solvation": -0.006719,
+    }, id='fcm'),
+    #
+    pytest.param({
+        "geom": __geoms["fcm"],
+        "ddx": {"solvent_epsilon": 2.0, "eta": 0, },
+        "ref": -594.990420855,  # from Gaussian
+        "solvation": -0.002964,
+    }, id='fcmeps'),
+    #
+    pytest.param({
+        "geom": __geoms["fcm"],
+        "ddx": {"solvent_epsilon": 2.0, "eta": 0.2, },
+        "ref": -594.990487330,  # from Gaussian
+        "solvation": -0.003041,
+    }, id='fcmepseta'),
+    #
+    pytest.param({
+        "geom": __geoms["benzene"],
+        "ddx": {"solvent_epsilon": 1e8, "eta": 0, },
+        "ref": -229.420391688,  # from Gaussian
+        "solvation": -0.005182,
+    }, id='benzene'),
+])
+def test_ddx_rhf_reference(inp):
+    mol = psi4.geometry(inp["geom"])
+    psi4.set_options({
+        "ddx": True,
+        "basis": "3-21g",
+        "guess": "core",
+        "scf_type": "direct",
+        #
+        "ddx__model": "cosmo",
+        "ddx__radii_set": "uff",
+        "ddx__lmax": 3,
+        "ddx__n_lebedev": 302,
```

Comment:
isn't the lebedev number 29 for sph 302? should both options be available?

Reply:
I see. Indeed that might be confusing. What you refer to is the Lebedev order, but what is meant here (and in the appropriate DDX option) is the number of actual points used to discretise the unit sphere, which for Lebedev order 29 is 302. I've used this flag for consistency with the pyddx interface.